bundles/org.openhab.binding.linky/src/main/java/org/openhab/binding/linky/internal/LinkyHandlerFactory.java

```diff
@@ -80,9 +120,16 @@ public LinkyHandlerFactory(final @Reference LocaleProvider localeProvider,
         } catch (KeyManagementException e) {
             logger.warn("An exception occurred while initialising the SSL context : '{}'", e.getMessage(), e);
         }
+        this.thingRegistry = thingRegistry;
```

Comment:
I was wondering what you did with the `thingRegistry` in the `HandlerFactory`. Reading what comes below (fallback for MyElectrical data) - I'm convinced this has to be changed. You give too much responsibilities to `HandlerFactory`. You could take a look at how oAuth granting is done in other bindings (probably not in the HandlerFactory). Netatmo binding can be an example.

Reply:
I refactor LinkyHandlerFactory, moving oAuth stuff to LinkyHandler

Reply:
Sorry, this is move to ApiBridgeHandler, not LinkyHandler !